src/Extensions/Hosting.Services/Internal/OmexStatefulServiceRunner.cs

```diff
@@ -3,60 +3,29 @@
 
 using System.Collections.Generic;
 using System.Fabric;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Hosting;
-using Microsoft.ServiceFabric.Services.Communication.Runtime;
 using Microsoft.ServiceFabric.Services.Runtime;
 
 namespace Microsoft.Omex.Extensions.Hosting.Services
 {
-	/// <summary>
-	/// Class to manage stateful service fabric service startup
-	/// </summary>
-	internal sealed class OmexStatefulServiceRunner : IOmexServiceRunner
+	internal sealed class OmexStatefulServiceRunner : OmexServiceRunner<OmexStatefulService, StatefulServiceContext>
 	{
-		private readonly string m_applicationName;
-		private readonly ServiceContextAccessor<StatefulServiceContext> m_contextAccessor;
-		private IEnumerable<IListenerBuilder<StatefulServiceContext>> ListenerBuilders { get; }
-		private IEnumerable<IServiceAction<StatefulServiceContext>> ServiceActions { get; }
-
 		public OmexStatefulServiceRunner(
 			IHostEnvironment environment,
 			ServiceContextAccessor<StatefulServiceContext> contextAccessor,
-			IEnumerable<IListenerBuilder<StatefulServiceContext>> listenerBuilders,
-			IEnumerable<IServiceAction<StatefulServiceContext>> serviceActions)
-		{
-			m_applicationName = environment.ApplicationName;
-			m_contextAccessor = contextAccessor;
-			ListenerBuilders = listenerBuilders;
-			ServiceActions = serviceActions;
-		}
+			IEnumerable<IListenerBuilder<OmexStatefulService>> listenerBuilders,
+			IEnumerable<IServiceAction<OmexStatefulService>> serviceActions)
+				: base(environment, contextAccessor, listenerBuilders, serviceActions) { }
 
-		public Task RunServiceAsync(CancellationToken cancellationToken) =>
-			ServiceRuntime.RegisterServiceAsync(m_applicationName, ServiceFactory, cancellationToken: cancellationToken);
+		public override Task RunServiceAsync(CancellationToken cancellationToken) =>
+			ServiceRuntime.RegisterServiceAsync(ApplicationName, ServiceFactory, cancellationToken: cancellationToken);
```

Comment:
SF expects the service type name here, is it passed from outside?

Reply:
Yes. when calling `BuildeStatefulService` or `BuildStatelessService` you provide an application name for Generic Host that will be used as a service name.